app/services/BusinessReportingMethodService.scala

```diff
@@ -0,0 +1,158 @@
+
+@Singleton
+class BusinessReportingMethodService @Inject()(incomeTaxViewChangeConnector: IncomeTaxViewChangeConnector,
+                                               dateService: DateService,
+                                               implicit val appConfig: FrontendAppConfig) extends FeatureSwitching {
+
+  def checkITSAStatusCurrentYear(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Boolean] = {
+    val yearEnd = dateService.getCurrentTaxYearEnd(isEnabled(TimeMachineAddYear)).toString.substring(2).toInt
+    val yearStart = yearEnd - 1
+
+    incomeTaxViewChangeConnector.getITSAStatusDetail(
+      nino = user.nino,
+      taxYear = s"$yearStart-$yearEnd",
+      futureYears = false,
+      history = false).flatMap {
+      case Right(itsaStatus) =>
+        val isMandatedOrVoluntary = itsaStatus.exists(_.itsaStatusDetails.exists(_.exists(_.isMandatedOrVoluntary)))
+        Future.successful(isMandatedOrVoluntary)
+      case Left(x: ITSAStatusResponseError) =>
+        Logger("application").error(s"[BusinessReportingMethodService][checkITSAStatusCurrentYear] $x")
+        Future.failed(new InternalServerException("[BusinessReportingMethodService][checkITSAStatusCurrentYear] - Failed to retrieve ITSAStatus"))
+    }
+  }
+
+  def isTaxYearCrystallised(ty: Int)(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Boolean]] = {
+    if (ty <= 2023) {
+      incomeTaxViewChangeConnector.getLegacyCalculationList(Nino(user.nino), ty.toString).flatMap {
+        case res: CalculationListModel => Future.successful(res.crystallised)
+        case err: CalculationListErrorModel => Future.failed(new InternalServerException(err.message))
+      }
+    } else {
+      val tyRange = s"${(ty - 1).toString.substring(2)}-${ty.toString.substring(2)}"
+      incomeTaxViewChangeConnector.getCalculationList(Nino(user.nino), tyRange).flatMap {
+        case res: CalculationListModel => Future.successful(res.crystallised)
+        case err: CalculationListErrorModel => Future.failed(new InternalServerException(err.message))
+      }
+    }
+
+  }
+
+  def getBusinessReportingMethodDetails(incomeSourceId: String)(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Option[BusinessReportingMethodViewModel]] = {
+    incomeTaxViewChangeConnector.getBusinessDetails(user.nino).flatMap {
+      case value: IncomeSourceDetailsModel =>
+        val latencyDetails: Option[LatencyDetails] = value.businesses.find(_.incomeSourceId.getOrElse("").equals(incomeSourceId)).flatMap(_.latencyDetails)
+        latencyDetails match {
+          case Some(x) =>
+            val currentTaxYearEnd = dateService.getCurrentTaxYearEnd(isEnabled(TimeMachineAddYear))
+            x match {
+              case LatencyDetails(_, _, _, ty2, _) if ty2.toInt < currentTaxYearEnd => Future.successful(None)
+              case LatencyDetails(_, ty1, ty1i, ty2, ty2i) =>
```

Comment:
can we be a bit more verbose? eg  ty => taxYear, ty1i => taxYearLatencyIndicator